### PR TITLE
Improve error messages while downcasting Int64Array

### PR DESCRIPTION
**Pull request by retikulum**
# Which issue does this PR close?

Part of #3152.

# Rationale for this change

It is very clear in the issue but there are different schemas while downcasting an ArrayRef to a related arrow array type. This is the new PR of improving downcasting of `Int64Array`.
# What changes are included in this PR?

- `as_int64_array` function is implemented in `datafusion\common\src\cast.rs`
- Refactor necessary parts

# Are there any user-facing changes?

I don't think so.

### CODE REVIEW

File: datafusion/core/tests/user_defined_plan.rs [hunk 1]
```diff
@@ -551,11 +552,7 @@ fn accumulate_batch(
         .downcast_ref::<StringArray>()
         .expect("Column 0 is not customer_id");
 
-    let revenue = input_batch
-        .column(1)
-        .as_any()
-        .downcast_ref::<Int64Array>()
-        .expect("Column 1 is not revenue");
+    let revenue = as_int64_array(input_batch.column(1)).unwrap();
```
Comment (viirya) [inline datafusion/core/tests/user_defined_plan.rs:555]: I think that you can still keep `.expect("Column 1 is not revenue");`?

Comment (retikulum) [reply datafusion/core/tests/user_defined_plan.rs:555]: I agree but I am not sure if we will give specific error to user. If it is the preferred way, I will change it

Comment (viirya) [reply datafusion/core/tests/user_defined_plan.rs:555]: As this is test code, I think if the test data is correct, it must return a Int64Array for column 1. `Column 1 is not revenue` as error message, I think its purpose is to indicate that.

Both (previous and this PR) are fine. I'd keep it but it looks okay too for this change.

**Review by viirya — APPROVED** [review]

File: datafusion/core/tests/user_defined_plan.rs (hunk 1 above)
Comment (alamb) [reply datafusion/core/tests/user_defined_plan.rs:555]: Yeah I agree with @viirya

File: datafusion/core/src/datasource/file_format/json.rs [hunk 2]
```diff
@@ -228,11 +228,7 @@ mod tests {
         assert_eq!(1, batches[0].num_columns());
         assert_eq!(12, batches[0].num_rows());
 
-        let array = batches[0]
-            .column(0)
-            .as_any()
-            .downcast_ref::<Int64Array>()
-            .unwrap();
+        let array = as_int64_array(batches[0].column(0))?;
```
Comment (alamb) [inline datafusion/core/src/datasource/file_format/json.rs:231]: this is so much nicer

**Review by alamb — APPROVED** [review]
Thanks @retikulum !

**Pull request merged**